securedrop_client/api_jobs/base.py

```diff
@@ -30,29 +30,35 @@ class ApiJob(QObject):
     '''
     failure_signal = pyqtSignal(Exception)
 
-    def __init__(self) -> None:
+    def __init__(self, remaining_attempts: int = 5) -> None:
         super().__init__(None)  # `None` because the QOjbect has no parent
+        self.remaining_attempts = remaining_attempts
 
     def _do_call_api(self, api_client: API, session: Session) -> None:
         if not api_client:
             raise ApiInaccessibleError()
 
-        try:
-            result = self.call_api(api_client, session)
-        except AuthError as e:
-            raise ApiInaccessibleError() from e
-        except ApiInaccessibleError as e:
-            # Do not let ApiInaccessibleError emit the regular failure signal, raise now
-            raise ApiInaccessibleError() from e
-        except RequestTimeoutError:
-            logger.debug('Job {} timed out'.format(self))
-            raise
-        except Exception as e:
-            logger.error('Job {} raised an exception: {}: {}'
-                         .format(self, type(e).__name__, e))
-            self.failure_signal.emit(e)
-        else:
-            self.success_signal.emit(result)
+        while self.remaining_attempts >= 1:
+            try:
+                self.remaining_attempts -= 1
+                result = self.call_api(api_client, session)
+            except AuthError as e:
+                raise ApiInaccessibleError() from e
+            except ApiInaccessibleError as e:
+                # Do not let ApiInaccessibleError emit the regular failure signal, raise now
+                raise ApiInaccessibleError() from e
+            except RequestTimeoutError:
+                logger.debug('Job {} timed out'.format(self))
+                if self.remaining_attempts == 0:
+                    raise
+            except Exception as e:
+                logger.error('Job {} raised an exception: {}: {}'
+                             .format(self, type(e).__name__, e))
+                self.failure_signal.emit(e)
```

Comment:
I think we need to emit the failure_signal after reaching the max retries so the client can respond with an error message.

Reply:
My thinking here was that I should keep the current signal/slot behavior the same as master since we discussed having a new signal for RequestTimeoutError that would pull up the help screen indicating the user should fix their network (the other half of https://github.com/freedomofpress/securedrop-client/issues/379). Then the cached job queue logic should re-run the job when network re-connects and then the failure/success signal would be emitted naturally